system/jlib/jfile.cpp

```diff
@@ -245,11 +245,12 @@ static StringBuffer &getLocalOrRemoteName(StringBuffer &name,const RemoteFilenam
 CFile::CFile(const char * _filename)
 {
     filename.set(_filename);
-    flags = ((unsigned)IFSHread)|((S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH)<<16);
+    flags = ((unsigned)IFSHread)|((S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)<<16);
 }
 
 void CFile::setCreateFlags(unsigned cflags)
 {
+    flags &= 0xffff;
```

Comment:
see other comments - afaics, setCreateFlags should be changed to take an unsigned short.